rpm/0070-sailfishos-webrtc-Implement-video-capture-module.-JB.patch

```diff
@@ -0,0 +1,2237 @@
++ */
++
++#ifndef WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_SFOS_DEVICE_INFO_SFOS_H_
++#define WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_SFOS_DEVICE_INFO_SFOS_H_
++
++#include "webrtc/modules/video_capture/device_info_impl.h"
++#include "webrtc/modules/video_capture/video_capture_impl.h"
++#include "webrtc/base/platform_thread.h"
++
++#include "webrtc/modules/video_capture/sfos/camera_manager.h"
++
++namespace webrtc
++{
++namespace videocapturemodule
++{
++class DeviceInfoSFOS: public DeviceInfoImpl
++{
++public:
++    DeviceInfoSFOS();
++    virtual ~DeviceInfoSFOS();
++    virtual uint32_t NumberOfDevices();
++    virtual int32_t GetDeviceName(
++        uint32_t deviceNumber,
++        char* deviceNameUTF8,
++        uint32_t deviceNameLength,
++        char* deviceUniqueIdUTF8,
++        uint32_t deviceUniqueIdUTF8Length,
++        char* productUniqueIdUTF8=0,
++        uint32_t productUniqueIdUTF8Length=0,
++        pid_t* pid=0);
++    virtual int32_t CreateCapabilityMap (const char* deviceUniqueIdUTF8);
++    virtual int32_t DisplayCaptureSettingsDialogBox(
++        const char* /*deviceUniqueIdUTF8*/,
++        const char* /*dialogTitleUTF8*/,
++        void* /*parentWindow*/,
++        uint32_t /*positionX*/,
++        uint32_t /*positionY*/) { return -1; }
++    void FillCapabilities(const char* devName);
++    int32_t Init();
++private:
++    void FindCameras();
++    std::vector<CameraInfo> _deviceList;
++    std::map<std::string, camera_type_t> _deviceMap;
++    static std::vector<camera_type_t> supportedCameraTypes();
++};
++}  // namespace videocapturemodule
++}  // namespace webrtc
++#endif // WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_SFOS_DEVICE_INFO_SFOS_H_
+diff --git a/media/webrtc/trunk/webrtc/modules/video_capture/sfos/gst_droid_camera.cc b/media/webrtc/trunk/webrtc/modules/video_capture/sfos/gst_droid_camera.cc
+new file mode 100644
+index 000000000000..06df7a46f31f
+--- /dev/null
++++ b/media/webrtc/trunk/webrtc/modules/video_capture/sfos/gst_droid_camera.cc
+@@ -0,0 +1,490 @@
++/*
++ *  Copyright (c) 2021 Open Mobile Platform LLC. All Rights Reserved.
++ *
++ *  Use of this source code is governed by a BSD-style license
++ *  that can be found in the LICENSE file in the root of the source
++ *  tree. An additional intellectual property rights grant can be found
++ *  in the file PATENTS.  All contributing project authors may
++ *  be found in the AUTHORS file in the root of the source tree.
++ */
++
++#include <errno.h>
++#include <fcntl.h>
++#include <stdio.h>
++#include <string.h>
++#include <sys/ioctl.h>
++#include <sys/mman.h>
++#include <sys/stat.h>
++#include <unistd.h>
++
++#include <new>
++
++#include "webrtc/base/refcount.h"
++#include "webrtc/base/scoped_ref_ptr.h"
++#include "webrtc/system_wrappers/include/trace.h"
++
++#include "webrtc/modules/video_capture/sfos/gst_droid_camera.h"
++
++namespace webrtc {
++namespace videocapturemodule {
++
++#define GST_DROID_CAMERA_PRIMARY   "primary"
```

Comment:
Are these still valid? I think we have to use the camera number now

Reply:
Yes, this is reported as camera ID, see `GstDroidCameraManager::GetDeviceName()` below.